src/util/extent.js

```diff
@@ -1,57 +1,105 @@
 import d3 from 'd3';
+import {noop} from './fn';
 
 /**
  * The extent function enhances the functionality of the equivalent D3 extent function, allowing
  * you to pass an array of fields, or accessors, which will be used to derive the extent of the supplied array. For
  * example, if you have an array of items with properties of 'high' and 'low', you
- * can use <code>fc.util.extent(data, ['high', 'low'])</code> to compute the extent of your data.
+ * can use <code>fc.util.extent()(data, ['high', 'low'])</code> to compute the extent of your data.
  *
  * @memberof fc.util
- * @param {array} data an array of data points, or an array of arrays of data points
- * @param {array} fields the names of object properties that represent field values, or accessor functions.
  */
-export default function(data, fields) {
-
-    // we need an array of arrays if we don't have one already
-    if (!Array.isArray(data[0])) {
-        data = [data];
-    }
-    if (arguments.length === 2) {
-        // the fields parameter must be an array of field names, but we can pass non-array types in
-        if (!Array.isArray(fields)) {
-            fields = [fields];
+export default function() {
+
+    var include = noop,
+        pad = d3.functor(0);
```

Comment:
Neither `include` or `pad` need to be functions as you never pass any arguments to them when invoked.

We use functions elsewhere for things like property accessors, and use `d3.functor` to allow someone to supply a constant, e.g.

```
foo.xValue(function(d, i) { return 0; })
```

Can be simplified to:

```
foo.xValue(0)
```

thanks to `d3.functor`